### PR TITLE
Add `get_unchecked` method to `JsString` and `JsStr`

### PR DESCRIPTION
**Pull request by CrazyboyQCD**
This adds `get_unchecked` method to `JsString` and `JsStr` for future possible optimization usage.

### CODE REVIEW

File: core/string/src/str.rs [hunk 1]
```diff
@@ -307,6 +324,20 @@ impl<'a> JsSliceIndex<'a> for usize {
             JsStrVariant::Utf16(v) => v.get(index).copied(),
         }
     }
+
+    #[inline]
+    /// # Safety
+    ///
+    /// Caller must ensure the index is not out of bounds
+    unsafe fn get_unchecked(value: JsStr<'a>, index: Self) -> Self::Value {
```
Comment (jedel1043) [inline core/string/src/str.rs:332]: Nit: We prefer having doc comments above tags/derive macros.

File: core/string/src/str.rs [hunk 2]
```diff
@@ -191,6 +191,21 @@ impl<'a> JsStr<'a> {
         I::get(self, index)
     }
 
+    /// Returns an element or subslice depending on the type of index, without doing bounds check.
```
Comment (jedel1043) [inline core/string/src/str.rs:194]: This seems like a better doc comment than the one used for `JsString`s version of the method. Can you reuse this for that?

**Review by jedel1043 — COMMENTED** [review]
Implementation wise looks perfect! I just have a styling nitpick and a doc suggestion.

**Review by jedel1043 — APPROVED** [review]
Looks great! Thank you for your contribution!

**Review by HalidOdat — APPROVED** [review]
Looks good to me! :)

**Pull request merged**